### PR TITLE
Implemented Custom TextField Class and Refactored Onboarding Code

### PR DESCRIPTION
**Pull request by rs929**
## Overview

- Updated Cocoapods
- Implemented custom LabeledTextField class to abstract all labeled textField
- Refactored all Onboarding code to abstract out separate views into custom LabeledTextField
- Fixed functionality of textField so that labels only show up when there is text in the textField

## Screenshots (delete if not applicable)

Before/ After

### CODE REVIEW

File: Scoop/View/LabeledTextField.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+//  Created by Richie Sun on 9/27/23.
+//
+
+import UIKit
+
+class LabeledTextField: UIView {
+    
+    weak var delegate: UITextFieldDelegate?
+    
+    private let nameLabel = UILabel()
```
Comment (tiffany-pann) [inline Scoop/View/LabeledTextField.swift:16]: No // MARK :((

File: Scoop/View/LabeledTextField.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+            textField.layer.borderWidth = 1
+            textField.layer.borderColor = UIColor.black.cgColor
+        }
+    }
+    
+    func hidesLabel(isHidden: Bool) {
+        nameLabel.isHidden = isHidden
+    }
+    
+
```
Comment (tiffany-pann) [inline Scoop/View/LabeledTextField.swift]: Extra linebreak here

**Review by tiffany-pann — APPROVED** [review]
Looks good! I think it might be a good idea to make the placeholder text a little lighter, I was a little confused when I first watched the 'after' video because it looks like input rather than placeholder right now. Also does the placeholder reappear OK if you type some stuff and then delete it?

**Review by caitlynjin — APPROVED** [review]
I agree with what Tiffany said about lighter placeholder text, but everything else looks good!

File: Scoop/Controller/Onboarding/AboutYouViewController.swift [hunk 3]
```diff
@@ -12,15 +12,10 @@ class AboutYouViewController: OnboardingViewController {
...
-    private let yearTextField = OnboardingTextField()
-    
-    private let nameLabel = UILabel()
-    private let pronounsLabel = UILabel()
-    private let hometownLabel = UILabel()
-    private let yearLabel = UILabel()
+    private let nameTextField = LabeledTextField()
+    private let pronounsTextField = LabeledTextField()
+    private let hometownTextField = LabeledTextField()
+    private let yearTextField = LabeledTextField()
```
Comment (vinnie4k) [inline Scoop/Controller/Onboarding/AboutYouViewController.swift:18]: Sort these alphabetically.

File: Scoop/View/LabeledTextField.swift [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+import UIKit
+
+class LabeledTextField: UIView {
+    
+    weak var delegate: UITextFieldDelegate?
+    
+    private let nameLabel = UILabel()
+    let textField = OnboardingTextField()
+    
+    func setup(title: String, placeholder: String? = nil) {
```
Comment (vinnie4k) [inline Scoop/View/LabeledTextField.swift:21]: Add your `// MARK:` comments to keep it consistent with the codebase.

File: Scoop/View/LabeledTextField.swift [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        nameLabel.font = .systemFont(ofSize: 12)
+        nameLabel.textColor = .black
+        nameLabel.backgroundColor = .white
+        nameLabel.isHidden = true
+        addSubview(nameLabel)
+        
+        nameLabel.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(16)
+            make.centerY.equalTo(textField.snp.top)
+        }
```
Comment (vinnie4k) [inline Scoop/View/LabeledTextField.swift:64]: Same as above.

File: Scoop/View/LabeledTextField.swift [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        textField.layer.borderColor = UIColor.black.cgColor
+        textField.delegate = self.delegate
+        textField.isUserInteractionEnabled = true
+        textField.associatedView = self
+        addSubview(textField)
+        
+        textField.snp.makeConstraints { make in
+            make.leading.trailing.bottom.equalToSuperview()
+            make.height.equalTo(56)
+        }
```
Comment (vinnie4k) [inline Scoop/View/LabeledTextField.swift:50]: Personally, I would split this up into a helper function for readability.

File: Scoop/View/LabeledTextField.swift (hunk 2 above)
Comment (vinnie4k) [reply Scoop/View/LabeledTextField.swift]: +1

**Review by vinnie4k — APPROVED** [review]
Looks good! Made some comments I think you should address. Also, your screenshots aren't loading in the PR description (I would also provide more details in your description).

File: Scoop/Controller/Onboarding/AboutYouViewController.swift (hunk 3 above)
Comment (rs929) [reply Scoop/Controller/Onboarding/AboutYouViewController.swift:18]: They appear in the order they show up on the screen, so its easier to go back and edit

**Pull request merged**